lib/chef/provisioning/convergence_strategy/install_sh.rb

```diff
@@ -24,10 +24,10 @@ def initialize(convergence_options, config)
       attr_reader :bootstrap_env
 
       def setup_convergence(action_handler, machine)
-        super
-
         # Install chef-client.  TODO check and update version if not latest / not desired
         if machine.execute_always('chef-client -v').exitstatus != 0
+          super
+          
```

Comment:
My suggested change is this - do not perform `PrecreateChefObjects#setup_convergence` unless we know chef has not been installed.  If you like this change I'll need to copy it to the other convergence strategies.  My idea was to make these convergence methods idempotent.

\cc @chef/chef-metal-committers